pkg/apis/core/v1alpha1/uibutton_types.go

```diff
@@ -129,6 +139,16 @@ func (in *UIButton) Validate(_ context.Context) field.ErrorList {
 			locField.Child("componentType"), "Parent component type is required"))
 	}
 
+	if strings.EqualFold(string(in.Spec.Location.ComponentType), string(ComponentTypeGlobal)) {
+		switch strings.ToLower(in.Spec.Location.ComponentID) {
+		case "nav":
+			if in.Spec.Icon == "" {
+				fieldErrors = append(fieldErrors, field.Required(
```

Comment:
also - i think it would be way funnier to pick an icon at random, based on, say, a hash of the button text

Reply:
or always default to a goose icon or smth

Reply:
Ah that's fair, we can pick a default to make it friendlier - I'll remove this